src/content/datachannel/datatransfer/js/main.js

```diff
@@ -41,97 +43,78 @@ megsToSend.addEventListener('change', function(e) {
 
   localConnection = new RTCPeerConnection(servers);
-  console.log('Created local peer connection object localConnection');
 
+  // Let's make a data channel!
   const dataChannelParams = {ordered: false};
   if (orderedCheckbox.checked) {
     dataChannelParams.ordered = true;
   }
-
   sendChannel = localConnection.createDataChannel('sendDataChannel', dataChannelParams);
   sendChannel.binaryType = 'arraybuffer';
-  console.log('Created send data channel');
-
-  sendChannel.onopen = onSendChannelStateChange;
-  sendChannel.onclose = onSendChannelStateChange;
-  localConnection.onicecandidate = e => onIceCandidate(localConnection, e);
+  sendChannel.bufferedAmountLowThreshold = CHUNK_SIZE / 2;
+  sendChannel.addEventListener('open', onSendChannelStateChange);
+  sendChannel.addEventListener('close', onSendChannelStateChange);
+  console.log('Created send data channel: ', sendChannel);
 
-  localConnection.createOffer().then(gotDescription1, onCreateSessionDescriptionError);
+  console.log('Created local peer connection object localConnection: ', localConnection);
 
-  remoteConnection = remoteConnection = new RTCPeerConnection(servers);
-  console.log('Created remote peer connection object remoteConnection');
+  localConnection.addEventListener('icecandidate', e => onIceCandidate(localConnection, e));
 
-  remoteConnection.onicecandidate = e => onIceCandidate(remoteConnection, e);
-  remoteConnection.ondatachannel = receiveChannelCallback;
-}
+  remoteConnection = new RTCPeerConnection(servers);
+  remoteConnection.addEventListener('icecandidate', e => onIceCandidate(remoteConnection, e));
+  remoteConnection.addEventListener('datachannel', receiveChannelCallback);
 
-function onCreateSessionDescriptionError(error) {
-  console.log(`Failed to create session description: ${error.toString()}`);
+  try {
+    const localOffer = await localConnection.createOffer();
+    await handleLocalDescription(localOffer);
+  } catch (e) {
+    console.error('Failed to create session description: ', e);
+  }
 }
 
-function randomAsciiString(length) {
-  let result = '';
-  for (let i = 0; i < length; i++) {
-    // Visible ASCII chars are between 33 and 126.
-    result += String.fromCharCode(33 + Math.random() * 93);
+function sendData(e) {
+  if (sendProgress.value < sendProgress.max) {
+    // console.log('sendChannel.bufferedAmountLowThreshold:', sendChannel.bufferedAmountLowThreshold);
+    // console.log('Buffered amount before sending:', sendChannel.bufferedAmount);
+    sendChannel.send(DATA_STRING);
+    sendProgress.value += CHUNK_SIZE;
+    // console.log('Buffered amount after sending:', sendChannel.bufferedAmount);
+    // console.log(`Sent another ${CHUNK_SIZE} bytes!`);
+
+    // We need to use setTimeout() at the moment due to a bug with bufferedamountlow event.
+    // See https://bugs.chromium.org/p/chromium/issues/detail?id=878682 for details
+    // sendChannel.addEventListener('bufferedamountlow', sendData, {'once': true});
```

Comment:
i don't think it is as bad... the old strategy from 4c06b2ebf40f46d7969cb7b035f4dfa71731baf9 works afaics. Also this was originally written by bemasc who implemented bufferedamountlow. Main problem is that this behaviour might be according to an old version of the spec which has probably flipflopped.

Reply:
The comment is still valid though. Once it's fixed, we can change to this behavior.